Some tests need to be disabled to get consistent result between OpenSSL 3 and 1.
To allow these tests to pass we would need to install openssl-3/legacy-provider.

Should be reported to upstream.

--- xmlsec1-1.2.42/tests/testDSig.sh
+++ xmlsec1-1.2.42/tests/testDSig.sh
@@ -693,20 +693,6 @@ "sha512 hmac-sha512 kw-aes256" \
     "hmac aes" \
     "--keys-file $topfolder/merlin-xmlenc-five/keys.xml $url_map_xml_stylesheet_2005"
 
-execDSigTest $res_success \
-    "" \
-    "merlin-xmlenc-five/encsig-hmac-sha256-rsa-1_5" \
-    "sha1 hmac-sha256 rsa-1_5" \
-    "hmac rsa" \
-    "$priv_key_option $topfolder/merlin-xmlenc-five/rsapriv.$priv_key_format --pwd secret $url_map_xml_stylesheet_2005"
-
-execDSigTest $res_success \
-    "" \
-    "merlin-xmlenc-five/encsig-hmac-sha256-rsa-oaep-mgf1p" \
-    "sha1 hmac-sha256 rsa-oaep-mgf1p" \
-    "hmac rsa" \
-    "$priv_key_option $topfolder/merlin-xmlenc-five/rsapriv.$priv_key_format --pwd secret $url_map_xml_stylesheet_2005"
-
 
 
 ##########################################################################
--- xmlsec1-1.2.42/tests/testEnc.sh
+++ xmlsec1-1.2.42/tests/testEnc.sh
@@ -186,22 +186,6 @@ "--keys-file $topfolder/merlin-xmlenc-fi
 
 execEncTest $res_success \
     "" \
-    "merlin-xmlenc-five/encrypt-element-aes128-cbc-rsa-1_5" \
-    "aes128-cbc rsa-1_5" \
-    "$priv_key_option $topfolder/merlin-xmlenc-five/rsapriv.$priv_key_format --pwd secret" \
-    "--keys-file $topfolder/merlin-xmlenc-five/keys.xml --session-key aes-128 $priv_key_option $topfolder/merlin-xmlenc-five/rsapriv.$priv_key_format --xml-data $topfolder/merlin-xmlenc-five/encrypt-element-aes128-cbc-rsa-1_5.data --node-id Purchase --pwd secret"  \
-    "$priv_key_option $topfolder/merlin-xmlenc-five/rsapriv.$priv_key_format --pwd secret"
-
-execEncTest $res_success \
-    "" \
-    "merlin-xmlenc-five/encrypt-data-tripledes-cbc-rsa-oaep-mgf1p" \
-    "tripledes-cbc rsa-oaep-mgf1p" \
-    "$priv_key_option $topfolder/merlin-xmlenc-five/rsapriv.$priv_key_format --pwd secret" \
-    "--keys-file $topfolder/merlin-xmlenc-five/keys.xml --session-key des-192 $priv_key_option $topfolder/merlin-xmlenc-five/rsapriv.$priv_key_format --binary-data $topfolder/merlin-xmlenc-five/encrypt-data-tripledes-cbc-rsa-oaep-mgf1p.data --pwd secret"  \
-    "$priv_key_option $topfolder/merlin-xmlenc-five/rsapriv.$priv_key_format --pwd secret"
-
-execEncTest $res_success \
-    "" \
     "merlin-xmlenc-five/encrypt-data-aes256-cbc-kw-tripledes" \
     "aes256-cbc kw-tripledes" \
     "--keys-file $topfolder/merlin-xmlenc-five/keys.xml" \
@@ -256,69 +240,6 @@ # 01-phaos-xmlenc-3
 #
 ##########################################################################
 
-execEncTest $res_success \
-    "" \
-    "01-phaos-xmlenc-3/enc-element-3des-kt-rsa1_5" \
-    "tripledes-cbc rsa-1_5" \
-    "$priv_key_option $topfolder/01-phaos-xmlenc-3/rsa-priv-key.$priv_key_format --pwd secret" \
-    "--session-key des-192 --keys-file $topfolder/01-phaos-xmlenc-3/keys.xml --enabled-key-data key-name --xml-data $topfolder/01-phaos-xmlenc-3/enc-element-3des-kt-rsa1_5.data --node-name http://example.org/paymentv2:CreditCard"  \
-    "$priv_key_option $topfolder/01-phaos-xmlenc-3/rsa-priv-key.$priv_key_format --pwd secret"
-
-execEncTest $res_success \
-    "" \
-    "01-phaos-xmlenc-3/enc-element-3des-kt-rsa_oaep_sha1" \
-    "tripledes-cbc rsa-oaep-mgf1p" \
-    "$priv_key_option $topfolder/01-phaos-xmlenc-3/rsa-priv-key.$priv_key_format --pwd secret" \
-    "--session-key des-192 --keys-file $topfolder/01-phaos-xmlenc-3/keys.xml --enabled-key-data key-name --xml-data $topfolder/01-phaos-xmlenc-3/enc-element-3des-kt-rsa_oaep_sha1.data --node-name http://example.org/paymentv2:CreditCard"  \
-    "$priv_key_option $topfolder/01-phaos-xmlenc-3/rsa-priv-key.$priv_key_format --pwd secret"
-
-execEncTest $res_success \
-    "" \
-    "01-phaos-xmlenc-3/enc-element-aes128-kt-rsa1_5" \
-    "aes128-cbc rsa-1_5" \
-    "$priv_key_option $topfolder/01-phaos-xmlenc-3/rsa-priv-key.$priv_key_format --pwd secret" \
-    "--session-key aes-128 --keys-file $topfolder/01-phaos-xmlenc-3/keys.xml --enabled-key-data key-name --xml-data $topfolder/01-phaos-xmlenc-3/enc-element-aes128-kt-rsa1_5.data --node-name http://example.org/paymentv2:CreditCard"  \
-    "$priv_key_option $topfolder/01-phaos-xmlenc-3/rsa-priv-key.$priv_key_format --pwd secret"
-
-execEncTest $res_success \
-    "" \
-    "01-phaos-xmlenc-3/enc-element-aes128-kt-rsa_oaep_sha1" \
-    "aes128-cbc rsa-oaep-mgf1p" \
-    "$priv_key_option $topfolder/01-phaos-xmlenc-3/rsa-priv-key.$priv_key_format --pwd secret" \
-    "--session-key aes-128 --keys-file $topfolder/01-phaos-xmlenc-3/keys.xml --enabled-key-data key-name --xml-data $topfolder/01-phaos-xmlenc-3/enc-element-aes128-kt-rsa_oaep_sha1.data --node-name http://example.org/paymentv2:CreditCard"  \
-    "$priv_key_option $topfolder/01-phaos-xmlenc-3/rsa-priv-key.$priv_key_format --pwd secret"
-
-execEncTest $res_success \
-    "" \
-    "01-phaos-xmlenc-3/enc-element-aes192-kt-rsa_oaep_sha1" \
-    "aes192-cbc rsa-oaep-mgf1p" \
-    "$priv_key_option $topfolder/01-phaos-xmlenc-3/rsa-priv-key.$priv_key_format --pwd secret" \
-    "--session-key aes-192 --keys-file $topfolder/01-phaos-xmlenc-3/keys.xml --enabled-key-data key-name --xml-data $topfolder/01-phaos-xmlenc-3/enc-element-aes192-kt-rsa_oaep_sha1.data --node-name http://example.org/paymentv2:CreditCard"  \
-    "$priv_key_option $topfolder/01-phaos-xmlenc-3/rsa-priv-key.$priv_key_format --pwd secret"
-
-execEncTest $res_success \
-    "" \
-    "01-phaos-xmlenc-3/enc-text-aes192-kt-rsa1_5" \
-    "aes192-cbc rsa-1_5" \
-    "$priv_key_option $topfolder/01-phaos-xmlenc-3/rsa-priv-key.$priv_key_format --pwd secret" \
-    "--session-key aes-192 --keys-file $topfolder/01-phaos-xmlenc-3/keys.xml --enabled-key-data key-name --xml-data $topfolder/01-phaos-xmlenc-3/enc-text-aes192-kt-rsa1_5.data --node-name http://example.org/paymentv2:CreditCard"  \
-    "$priv_key_option $topfolder/01-phaos-xmlenc-3/rsa-priv-key.$priv_key_format --pwd secret"
-
-execEncTest $res_success \
-    "" \
-    "01-phaos-xmlenc-3/enc-content-aes256-kt-rsa1_5" \
-    "aes256-cbc rsa-1_5" \
-    "$priv_key_option $topfolder/01-phaos-xmlenc-3/rsa-priv-key.$priv_key_format --pwd secret" \
-    "--session-key aes-256 --keys-file $topfolder/01-phaos-xmlenc-3/keys.xml --enabled-key-data key-name --xml-data $topfolder/01-phaos-xmlenc-3/enc-content-aes256-kt-rsa1_5.data --node-name http://example.org/paymentv2:CreditCard"  \
-    "$priv_key_option $topfolder/01-phaos-xmlenc-3/rsa-priv-key.$priv_key_format --pwd secret"
-
-execEncTest $res_success \
-    "" \
-    "01-phaos-xmlenc-3/enc-text-aes256-kt-rsa_oaep_sha1" \
-    "aes256-cbc rsa-oaep-mgf1p" \
-    "$priv_key_option $topfolder/01-phaos-xmlenc-3/rsa-priv-key.$priv_key_format --pwd secret" \
-    "--session-key aes-256 --keys-file $topfolder/01-phaos-xmlenc-3/keys.xml --enabled-key-data key-name --xml-data $topfolder/01-phaos-xmlenc-3/enc-text-aes256-kt-rsa_oaep_sha1.data --node-name http://example.org/paymentv2:CreditCard"  \
-    "$priv_key_option $topfolder/01-phaos-xmlenc-3/rsa-priv-key.$priv_key_format --pwd secret"
 
 execEncTest $res_success \
     "" \
